### PR TITLE
Allow constructing BuildConfig with known deps

### PR DESCRIPTION
**Pull request by natebosch**
Rather than reparse the Pubspec add a constructor that takes know
package name and dependencies.

Don't export `Pubspec` class and delete `BuildConfigSet` since they
don't have use cases outside this package.

### CODE REVIEW

File: build_config/lib/src/build_config.dart [hunk 1]
```diff
@@ -58,18 +58,34 @@ class BuildConfig {
   static const _isOptional = 'is_optional';
   static const _buildTo = 'build_to';
 
-  /// Returns a parsed [BuildConfig] file in [path], if one exists.
+  /// Returns a parsed [BuildConfig] file in [path], if one exist, otherwise a
+  /// default config.
   ///
-  /// Otherwise uses the default setup.
-  static Future<BuildConfig> fromPackageDir(Pubspec pubspec, String path,
+  /// [path] must be a directory chich contains a `pubspec.yaml` file and
```
Comment (jakemac53) [inline build_config/lib/src/build_config.dart]: which

File: build_config/lib/src/build_config.dart [hunk 2]
```diff
@@ -58,18 +58,34 @@ class BuildConfig {
...
+    final pubspec = await Pubspec.fromPackageDir(path);
+    return fromBuildConfigDir(
+        pubspec.pubPackageName, pubspec.dependencies, path,
+        includeWebSources: includeWebSources);
+  }
+
+  /// Returns a parsed [BuildConfig] file in [path], if one exists, otherwise a
+  /// default config.
+  ///
+  /// [path] maybe be a directory which contains a `build.yaml`.
```
Comment (jakemac53) [inline build_config/lib/src/build_config.dart]: I would reword this a bit, maybe "[path] should be the path to a directory which may contain a `build.yaml`"

Comment (jakemac53) [reply build_config/lib/src/build_config.dart]: We could also consider checking that this is a Directory and throwing a nice error if not

File: build_config/lib/src/build_config.dart (hunk 1 above)
Comment (natebosch) [reply build_config/lib/src/build_config.dart]: Done

File: build_config/lib/src/build_config.dart (hunk 2 above)
Comment (natebosch) [reply build_config/lib/src/build_config.dart]: I think for now I'll avoid the extra check. I'm not sure if we're depending on allowing bad input. I can follow up with a later change.

File: build_config/lib/src/build_config.dart [hunk 3]
```diff
@@ -74,7 +74,7 @@ class BuildConfig {
   /// Returns a parsed [BuildConfig] file in [path], if one exists, otherwise a
   /// default config.
   ///
-  /// [path] maybe be a directory which contains a `build.yaml`.
+  /// [path] shouldd the path to a directory which may contain a `build.yaml`.
```
Comment (jakemac53) [inline build_config/lib/src/build_config.dart]: should be

Comment (natebosch) [reply build_config/lib/src/build_config.dart]: You'd think typing wouldn't be the hard part...
